test/no-dead-hash-links-within-document.md

```diff
@@ -0,0 +1,17 @@
+# Valid/Invalid Hash Links within document
+
+## Valid links
+
+[Valid link](#validinvalid-hash-links-within-document)
+
+[Valid link](#valid-links)
+
+[Valid link](#valid-h3-heading)
+
+### Valid h3 heading
+
+content...
+
+## Invalid links
+
+[Invalid link](#dead-slug-no-heading) {MD051}
```

Comment:
Please add a few more test cases. Maybe some that verify symbols get replaced. Let's be sure ATX headings are handled correctly.

Reply:
I see some new cases, thanks. Please include the two other heading kinds as well:

```markdown
Heading
=======

# Heading #
```